#!/usr/bin/env python
PACKAGE = "two_d_guidance"
    
from dynamic_reconfigure.parameter_generator_catkin import *
    
gen = ParameterGenerator()

# Pure Pursuit
lookahead_mode_enum = gen.enum([ gen.const("dist",  int_t, 0, "Dist"),
                                 gen.const("time",  int_t, 1, "Time")],
                               "lookahead_mode")
gen.add("lookahead_mode", int_t, 0, "lookahead_mode", 0, 0, 1, edit_method=lookahead_mode_enum)
gen.add("lookahead_dist",    double_t, 0, "Look Ahead Dist", .8,  0.01,   2.5)
gen.add("lookahead_time",    double_t, 0, "Look Ahead time", .5,  0.01,   1.5)

# Velocity
vel_ctl_mode_enum = gen.enum([ gen.const("cst",     int_t, 0, "Cst"),
                               gen.const("profile", int_t, 1, "Profile"),
                               gen.const("curv",    int_t, 2, "Curv")],
                             "vel ctl mode")
gen.add("vel_ctl_mode", int_t, 0, "vel_ctl_mode", 0, 0, 2, edit_method=vel_ctl_mode_enum)
gen.add("vel_sp",       double_t, 0, "Velocity setpoint", .4,  0.01,   5.)
gen.add("vel_k_curve",  double_t, 0, "Velocity curv coef", 0.5,  0.01,   2.5)

guidance_mode_enum = gen.enum([ gen.const("idle",   int_t, 0, "Idle"),
                                gen.const("stop",   int_t, 1, "Stop"),
                                gen.const("drive",  int_t, 2, "Drive")],
                               "guidance mode")
gen.add("guidance_mode", int_t, 0, "guidance mode", 0, 0, 2, edit_method=guidance_mode_enum)

exit(gen.generate(PACKAGE, "two_d_guidance", "trr_guidance"))
